runtime/testing/test-util.js

```diff
@@ -100,3 +100,12 @@ export async function assertSingletonIs(store, field, expectation) {
   const actual = await store.get();
   assert.equal(actual !== null ? actual.rawData[field] : '(null)', expectation);
 }
+
+// Example usage in async mocha tests:
+//   await until(() => eatery.latestOrder === 'Pizza');
+export function until(condition, checkIntervalMs = 5) {
```

Comment:
Is it possible for us to wait on a promise rather than looping on a timeout?

Reply:
Yes, thanks for prodding me in the right direction. Added a promise in the mock-slot-dom-consumer that allows to wait for the content to be set.